File: aeds2/tps/tp01/Is(recursivo)/Is.c
#include <stdio.h>

int x1(char *entrada, int cont){
	char c = entrada[cont];
	if(c == '\0' || c == '\n'){
		cont = 0;
		return 1;
	}
	else{
		if(c != 'a' && c != 'e' && c != 'i' && c != 'o' && c != 'u'){
			cont = 0;
			return 0;
		}
		else{
			return x1(entrada, cont + 1);
		}
	}
}

int x2(char *entrada, int cont){
	char c = entrada[cont];
	if(c == '\0' || c == '\n'){
		cont = 0;
		return 1;
	}
	else{
		if(!(c >= 'a' && c <= 'z') || (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u')){
			cont = 0;
			return 0;
		}
		else{
			return x2(entrada, cont + 1);
		}
	}
}

int x3(char *entrada, int cont){
	char c = entrada[cont];
	if(c == '\0' || c == '\n'){
		cont = 0;
		return 1;
	}
	else{
		if(!(c >= '0' && c <= '9')){
			cont = 0;
			return 0;
		}	
		else{
			return x3(entrada, cont + 1);
		}
	}
}

int x4(char *entrada, int cont){
	char c = entrada[cont];
	if(c == '\0' || c == '\n'){
		cont = 0;
		return 1;
	}
	else{
		if(!(c >= '0' && c <= '9') &&  c != '.'){
			cont = 0;
			return 0;
		}
		else{
			return x4(entrada, cont + 1);
		}
	}
}

int main(){
	char entrada[1000];
	int cont = 0;
	fgets(entrada, 1000, stdin);
	
	while(entrada[0] != 'F' || entrada[1] != 'I' || entrada[2] != 'M'){
		int i = 0;
		while(entrada[i] != '\0' && entrada[i] != '\n'){
			if(entrada[i] >= 'A' && entrada[i] <= 'Z'){
				entrada[i] = (entrada[i] + 32);
			}
			i++;
		}
		printf("%s", (x1(entrada, cont) == 1 ? "SIM " : "NAO "));
		printf("%s", (x2(entrada, cont) == 1 ? "SIM " : "NAO "));
		printf("%s", (x3(entrada, cont) == 1 ? "SIM " : "NAO "));
		printf("%s\n", (x4(entrada, cont) == 1 ? "SIM" : "NAO"));
		fgets(entrada, 1000, stdin);
	}

	return 0;
}

File: aeds2/tps/tp01/versao2/q03/Is.java
import java.util.Scanner;

public class Is{
	public static String conversor(String str){ //Converte maiúsculas em minúsculas
		String strMinuscula = "";
		for(int i = 0; i < str.length(); i++){
			char c = str.charAt(i);
			if(c >= 'A' && c <= 'Z'){
				strMinuscula += (char)(c + 32);
			}
			else{
				strMinuscula += c;
			}
		}
		return strMinuscula;
	}

	public static boolean x1(String str){ //Só vogais
		for(int i = 0; i < str.length(); i++){
			char c = str.charAt(i);
			if(c != 'a' && c != 'e' && c != 'i' && c != 'o' && c != 'u'){
				return false;
			}
		}
		return true;
	}
	
	public static boolean x2(String str){ //Só consoantes
		for(int i = 0; i < str.length(); i++){
			char c = str.charAt(i);
			if(!(c >= 'a' && c <= 'z') || (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u')){
				return false;
			}
		}
		return true;
	}

	public static boolean x3(String str){ //Número inteiro
		for(int i = 0; i < str.length(); i++){
			char c = str.charAt(i);
			if(!(c >= '0' && c <= '9')){
				return false;
			}
		}
		return true;
	}
	
	public static boolean x4(String str){ //Número real
		int cont = 0; //Verifica se existe apenas um ponto ou uma vírgula 
		for(int i = 0; i < str.length(); i++){
			char c = str.charAt(i);
			if(c == '.' || c == ','){
				cont++;
			}
			if((!(c >= '0' && c <= '9') && (c != '.' && c != ',')) || cont > 1){
				return false;
			}
		}
		return true;
	}


	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		String str = sc.nextLine();
		String strMinuscula;

		while(!str.equals("FIM")){
			strMinuscula = conversor(str);
			System.out.print(x1(strMinuscula) ? "SIM " : "NAO "); //Operador ternário para deixar o código mais limpo 
			System.out.print(x2(strMinuscula) ? "SIM " : "NAO ");
			System.out.print(x3(strMinuscula) ? "SIM " : "NAO ");
			System.out.println(x4(strMinuscula) ? "SIM" : "NAO");
			str = sc.nextLine();
		}		
	}
}
